### PR TITLE
Fix bug when updating existing goal sources

### PR DESCRIPTION
**Pull request by AdamAdHocTeam**
## Description of change

The migration modified by this PR has not run successfully on prod. The issue is because we already have goals and arg's using the source we are changing. This PR updates the way we handle this to first add the new ENUM switch it over and remove the old value.

**NOTE:** Verified that this migration has NOT run yet by querying SequelizeMeta table on latest prod db.
SELECT * FROM "SequelizeMeta" where "name" = '20240129221712-change-training-event-goal-source.js'

## How to test
- We should have the same number of Goals and Activity Report goals set with a  source value before and after.
- On a prod database run the migration make sure any goal source in the Goals or ActivityReportGoals that was using 'Training event follow-up' is now using 'Training event'. 

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-0

### Every PR

- [x] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Before merge to main

- [ ] OHS demo complete
- [ ] Ready to create production PR

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: src/migrations/20240129221712-change-training-event-goal-source.js [hunk 1]
```diff
@@ -8,28 +7,79 @@ module.exports = {
...
+      await queryInterface.sequelize.query(
+        `DELETE FROM pg_enum
+            WHERE enumlabel = 'Training event follow-up'
+              AND enumtypid = (
+                SELECT oid
+                FROM pg_type
+                WHERE typname = 'enum_Goals_source'
+              );`,
+        { transaction },
+      );
```
Comment (GarrettEHill) [inline src/migrations/20240129221712-change-training-event-goal-source.js:46]: This is not safe, do not do it this way

Comment (GarrettEHill) [inline src/migrations/20240129221712-change-training-event-goal-source.js:46]: This is not safe, do not do it this way. enums must only be added to or dropped

**Review by GarrettEHill — CHANGES_REQUESTED** [review]

File: src/migrations/20240129221712-change-training-event-goal-source.js (hunk 1 above)
Comment (GarrettEHill) [reply src/migrations/20240129221712-change-training-event-goal-source.js:46]: the pg_* tables are only for reading as they are generated. Do not edit them

Comment (AdamAdHocTeam) [reply src/migrations/20240129221712-change-training-event-goal-source.js:46]: What is the risk?

Comment (GarrettEHill) [reply src/migrations/20240129221712-change-training-event-goal-source.js:46]: This is fully unsupported, and not guaranteed to function as you are using it. the pg_* tables are for reading only not writing.  Please do it the correct way

Comment (GarrettEHill) [reply src/migrations/20240129221712-change-training-event-goal-source.js:46]: With enums, the only operations permitted are create, append, or drop.

Comment (GarrettEHill) [reply src/migrations/20240129221712-change-training-event-goal-source.js:46]: `ALTER TYPE mood_enum RENAME VALUE 'ok' TO 'content';`

Comment (AdamAdHocTeam) [reply src/migrations/20240129221712-change-training-event-goal-source.js:46]: Ok think I have the other way working doing one more test.

**Review by GarrettEHill — APPROVED** [review]

**Review by thewatermethod — APPROVED** [review]

**Pull request merged**